hw_9_corr.py

```diff
@@ -0,0 +1,287 @@
+# Cписок требований:
+# 1) Создайте товар с такими свойствами, как имя(name), подробные сведения(description or details),
+# количество на складе(quantity), доступность(availability), цена(price).
+# 2) Добавить товар на склад
+# 3) Удалить товар со склада
+# 4) Распечатать остаток товара по его имени
+# 5) Распечатать остаток всех товаров
+# 6) Товар может принадлежать к категории
+# 7) Распечатать список товаров с заданной категорией
+# 8) Корзина для покупок, в которой может быть много товаров с общей ценой.
+# 9) Добавить товары в корзину (вы не можете добавлять товары, если их нет в наличии)
+# 10) Распечатать элементы корзины покупок с ценой и общей суммой
+# 11) Оформить заказ и распечатать детали заказа по его номеру
+# 12) Позиция заказа, созданная после оформления заказа пользователем.
+# Он будет иметь идентификатор заказа(order_id), дату покупки(date_purchased), товары(items), количество(quantity)
+# 13) После оформления заказа количество товара уменьшается на количество товаров из заказа.
+
+
+# Добавить к этой задаче дескриптор для аттрибута цена.
+# При назначении цены товара будет автоматически добавлен НДС 20%
+# При получении цены товара, цена возврщается уже с учетом НДС
+
+
+class Material:
+    VAT = 0.20
+
+    class PriceDescriptor:
+        def __get__(self, instance, owner):
+            return instance.price * (1 + Material.VAT)
+
+    price = PriceDescriptor()
+
+    def __init__(self, name, details, quantity, availability, price):
+        self.name = name
+        self.details = details
+        self._quantity = quantity
+        self.availability = availability
+        self.price = price
+        self.category = None
+
+    @ property
+    def quantity(self):
+        return self._quantity
+
+    @ quantity.setter
+    def quantity(self, value):
+        self._quantity = value
+
+
+class Warehouse:
+    def __init__(self):
+        self.stock = []
+
+    def find_material(self, name):
+        if self.stock:
+            for material in self.stock:
+                if material.name == name:
+                    return material
+        return None
+
+    def add_material(self, name, details, quantity, availability, price):
+        material = Material(name, details, quantity, availability, price)
+        self.stock.append(material)
+
+    def remove_material(self, material):
+        material = self.find_material(material)
+        if material:
+            self.stock.remove(material)
+
+    def current_quantity(self, material):
+        material = self.find_material(material)
+        if material:
+            print('Current stock of {0} is {1}'.format(material.name, material.quantity))
+        else:
+            print('Material does not exist.')
+
+    def current_quantity_all(self):
+        if self.stock:
+            for material in self.stock:
+                self.current_quantity(material.name)
+        else:
+            print('Warehouse is empty.')
+
+
+class Category(Warehouse):
```

Comment:
why we need here Warehouse as a parent?

Reply:
Parent class Warehouse has been removed from Category Class.